zmapsdk/input.py

```
@@ -1,243 +1,243 @@
+        for ip in targets:
+            try:
+                ipaddress.ip_address(ip)
+            except ValueError as e:
+                raise ZMapInputError(f"Invalid IP address: {ip} - {e!s}")
+
+        # Write to file
+        try:
+            with open(output_file, "w") as f:
+                f.write("\n".join(targets))
+        except OSError as e:
+            raise ZMapInputError(f"Failed to create target file: {e!s}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)